### PR TITLE
random: add basic PCG32 RNG

### PR DESCRIPTION
**Pull request by ArsenArsen**

### CODE REVIEW

File: include/frg/random.hpp [hunk 1]
```diff
@@ -63,6 +64,43 @@ struct mt19937 {
...
+		state = 0;
+		inc = (seq << 1) | 1;
+		operator()();
+		state += seed;
+		operator()();
+	}
+
+	uint32_t operator()() {
+		uint64_t oldstate = this->state;
+		this->state = oldstate * 6364136223846793005ULL + this->inc;
```
Comment (qookei) [inline include/frg/random.hpp:85]: Same here, why not suffix the private members with a `_` to indicate that they're private members.

**Review by qookei — COMMENTED** [review]
Seems fine, just minor nitpicks

File: include/frg/random.hpp [hunk 2]
```diff
@@ -63,6 +64,43 @@ struct mt19937 {
 	int _ctr;
 };
 
+// Implementation based on https://github.com/imneme/pcg-c-basic
+// Copyright 2014 Melissa O'Neill <oneill@pcg-random.org>
+// Licensed under the Apache License, Version 2.0
+struct pcg_basic32 {
+	pcg_basic32(uint64_t seed, uint64_t seq = 1) {
+		this->seed(seed, seq);
```
[the remark this answers is not in the document]
Comment (ArsenArsen) [reply include/frg/random.hpp:72]: due to the parameter

File: include/frg/random.hpp (hunk 1 above)
Comment (ArsenArsen) [reply include/frg/random.hpp:85]: could do

File: include/frg/random.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (qookei) [reply include/frg/random.hpp:72]: Ah right

**Comment by avdgrinten** [comment]
Why does this class have `basic` in its name?

**Comment by ArsenArsen** [comment]
It's original library is `pcg_basic`, and it doesn't present all the flexibility of the PCG algorithm to the user

**Review by qookei — APPROVED** [review]

**Review by Geertiebear — APPROVED** [review]

**Pull request merged**